target/linux/ramips/dts/mt7628an_tplink_tl-wr902ac-v4.dts

```diff
@@ -0,0 +1,112 @@
+#include "mt7628an_tplink_8m.dtsi"
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "tplink,tl-wr902ac-v4", "mediatek,mt7628an-soc";
+	model = "TP-Link TL-WR902AC v4";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		button-reset {
+			label = "reset";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		button-sw1 {
+			label = "sw1";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+		};
+
+		button-sw2 {
+			label = "sw2";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_1>;
+		};
+
+		button-wps {
+			label = "wps";
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_lan {
+			label ="green:lan";
+            color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_LAN;
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_power: power {
+			label = "green:power";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_POWER;
+			gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
+		};
+
+		led_usb {
+			label = "green:usb";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_USB;
+			gpios = <&gpio 43 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		led_wan {
+			label = "green:wan";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WAN;
+			gpios = <&gpio 39 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan {
+			label = "green:wlan";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WLAN;
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wps {
+			label = "green:wps";
+			function = LED_FUNCTION_WPS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "i2s", "p0led_an", "p2led_an", "p4led_an", "uart1", "wdt", "wled_an";
+		function = "gpio";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x28000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+		nvmem-cells = <&macaddr_factory_f100>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <(-1)>;
```

Comment:
deprecated. ping @DragonBluep

Reply:
Any hints/doc how to change it?

Reply:
Hmm nvm. I was looking at https://github.com/openwrt/openwrt/commit/58502650e13d6634d57fe09520ac9773b5eb6dba but looks like I was mistaken.

Reply:
nvm: https://github.com/openwrt/openwrt/commit/0ba5c0bdc1ebddb50ab34806343d77baf54044be

Reply:
Original change:
br-lan    Link encap:Ethernet  HWaddr 40:ED:00:CF:B9:9B  
eth0      Link encap:Ethernet  HWaddr 40:ED:00:CF:B9:9B  
eth0.1    Link encap:Ethernet  HWaddr 40:ED:00:CF:B9:9B  
phy0-ap0  Link encap:Ethernet  HWaddr 40:ED:00:CF:B9:9B  
phy1-ap0  Link encap:Ethernet  HWaddr 40:ED:00:CF:B9:9A  

When I changed to : 
```
&pcie0 {
	mt76@0,0 {
		reg = <0x0000 0 0 0 0>;
		mediatek,mtd-eeprom = <&factory 0x28000>;
		ieee80211-freq-limit = <5000000 6000000>;
		nvmem-cells = <&macaddr_factory_f100 (-1)>;
		nvmem-cell-names = "mac-address";
	};
};
```
all addresses were 40:ED:00:CF:B9:9B (also  phy1-ap0) I am not sure if this is proper behavior. 
( <&macaddr_factory_f100 -1> did not compile)

Reply:
Nvmem layout change is needed too.

Reply:
Correct me if I am wrong:
This need to be changed in mt7628an_tplink_8m.dtsi file, but as this file is used in many places  I have 
1. create copy of dtsi (or use its contents directly) file and made changes,  OR
2. made much bigger PR with changes in all files used this dtsi
Taking into account that I have ver little (next to none) experience with dts it seems, that 1. is only option (or even close this PR)

Reply:
oh that is unfortunate. I guess this can stay as is.